### ISSUE TITLE
[Team 5] Economy

### ISSUE THREAD

**Issue by ryandmcneilly** [UQcsse3200/2023-studio-3#15]
# **Description**
Our team is tasked with developing the core mechanics of the in-game economy. The economy system will introduce strategic depth by influencing the player's ability to acquire new turrets, upgrade existing ones, and make strategic decisions. Players will earn currency through enemy eliminations and resource mining, adding a layer of strategy and engagement. Our responsibilities also include designing and implementing the UI elements that will display the player's income balance and rate.

## Sub Features/Component 
Each member will be responsible for one of the following components:
- Design of the UI (textures, sprites, images) 
- Implementing design onto game screen
- Income on mob drop (including graphic that you click)
- Design of economy and implementation of basic currency system (Abstraction class)
-  Implementing the turret that generates income (including design)

## **Examples**
- **UI**. The UI will display the amount of ‘scraps’ the user currently has and can spend.
- **scraps**: The main currency system, mobs will drop this currency on death. These scraps are used for the construction of turrets.
- **mine**: Alternative method of scrap generation. The user can place the mine on the grid, producing passive income which can be used for the construction of new turrets. The mine cannot attack mobs.
 
# **Dependencies**

This ticket relies on the completion of all subfeatures (task tickets):
- Basic Currency System (#19)
- UI Design (#16)
- UI Implementation (#18)
- Mob Drop (#28)
	- Implementing the mob drop relies on the basic class of mobs (#23)
- Income Turret (#30)
- Implementing the “income turret” relies on the basic class of turrets. (#5)

# **Milestones**
#### Goal 1 (Aug 20): Planning and Scaffolding 
- [x] Basic Currency System
- [x] UI Design
- [x] UI Implementation
- [ ] Mob Drop
- [ ] Income Turret
#### Goal 2 (Aug 23) Code Implementation and Test Design (JUnit)
- [x] Basic Currency System
- [ ] UI Implementation
- [ ] Mob Drop
- [ ] Income Turret 
#### Goal 3 (Aug 26) Testing and Debugging; Integration
- [ ] Basic Currency System
- [ ] UI Implementation
- [ ] Mob Drop
- [ ] Income Turret 
#### Goal 4 (Aug 28) Documentation
- [ ] Basic Currency System
- [ ] UI Implementation
- [ ] Mob Drop
- [ ] Income Turret 

**Completion Deadline:** Aug. 30

# **Extras**
These features are low priority and will be implemented if ahead of schedule:
- [ ] Second type of currency
- [ ] Special drop animation
- [ ] Sophisticated UI

# **Documentation**
- [Main description of feature](../wiki/EconomySystem)
- [User documentation](../wiki/Economy)

# **Members**
Ryan (@ryandmcneilly) (Ryan McNeilly)
Sonja (@SonjaMcNeilly) (Sonja)
Nawal (@nawal-0) (Nawal)
Jackie (@FattyHope) (Jackie) 
Kevin (@Hasakev) (Kevin Luo)

**Comment by Jonoym** [comment]
This feature ticket looks great! Just a reminder for everyone to make some task tickets to split up the work for the coming sprint.

**Comment by Mohamad11Dab** [comment]
Hey team 5, I'm working on the Tower class and plan to include all tower subclasses, including the income turret. Let me know if you'd like to handle the income turret implementation. I'll notify you once the Tower class is complete.

**Comment by ryandmcneilly** [comment]
Hey Mohammed, yes we would like to handle the income turret (which gives us a bit more work to do). Just give a heads-up whenever the Tower class is done. Cheers!
